packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -640,6 +650,15 @@ abstract class FlutterCommand extends Command<void> {
       }
     }
 
+    String analyzeSize;
+    if (argParser.options.containsKey(FlutterOptions.kAnalyzeSize)
+      && boolArg(FlutterOptions.kAnalyzeSize)
+      && !globals.platform.isWindows) {
+      final File file = globals.fsUtils.getUniqueFile(globals.fs.currentDirectory, 'flutter_size', '.json');
+      extraGenSnapshotOptions.add('--write-v8-snapshot-profile-to=${file.path}');
```

Comment:
can we generate a snapshot for each platform (see https://github.com/flutter/flutter/issues/61629)? we can write a follow up CL to consume all the generated snapshot files in SizeAnalyzer. Or do we actually need to generate different ABIs with the `flutter build apk` `--target-platform` flag?

Reply:
Though I'm fine with getting this change in, and adding support for multiple platforms later if that ends up being a CUJ.

Reply:
What do you mean by platform here - android/ios? or arm/arm-64/x86_64

Given the command UX we started with, I suspect only one platform at a time makes sense.

Reply:
I mean arm/arm-64/x86_64. One platform at a time sgtm

Reply:
I think we could do this as a follow up, but it would require more plumbing.